library/common/engine_handle.cc

```diff
@@ -0,0 +1,18 @@
+#include "library/common/engine_handle.h"
+
+// NOLINT(namespace-envoy)
+
+envoy_status_t EngineHandle::runOnEngineDispatcher(envoy_engine_t,
```

Comment:
instead of exposing this as C++ couldn't we have a `run_on_engine_dispatcher` c function that takes the envoy_engine_t and function pointer. Which in turn captures the engine ref instead of in the arg list?

What I am trying to get to is that now we are exposing functionality that is only available to C++ consumers.

Reply:
I think exposing C++ only functionality is kinda the point: it provides safe access to interacting with Envoy::Engine which is a C++ interface and wouldn't work in C.

How were you thinking a C function would interact with the engine? We either need to pass the Engine to the function or expose some way of retrieving a handle to the engine

I guess we could have a version that doesn't interact with the Engine but just runs some C code on the dispatcher, though I'm not sure what the use case would be

Reply:
I see. That makes sense.